app/models/assessment.rb

```diff
@@ -13,33 +13,33 @@ class Assessment < ApplicationRecord
   has_one :reported_condition, class_name: 'ReportedCondition'
   belongs_to :patient
 
-  scope :twenty_four_hours_fever_or_fever_medication, lambda {
-    where('created_at >= ?', 24.hours.ago).where_assoc_exists(:reported_condition, &:fever_or_fever_medication)
-  }
+  after_save :update_patient_linelist_after_save
+  before_destroy :update_patient_linelist_before_destroy
 
-  scope :seventy_two_hours_fever_or_fever_medication, lambda {
-    where('created_at >= ?', 72.hours.ago).where_assoc_exists(:reported_condition, &:fever_or_fever_medication)
-  }
-
-  scope :ten_days_symptomatic, lambda {
-    where('created_at >= ?', 10.days.ago).where(symptomatic: true)
-  }
-
-  scope :older_than_seventy_two_hours, lambda {
-    where('created_at <= ?', 72.hours.ago)
-  }
+  def update_patient_linelist_after_save
+    patient.symptom_onset = patient.assessments.where(symptomatic: true).minimum(:created_at)
```

Comment:
Isn't there a function on patient to handle this? I feel like there was more logic in there as well

Reply:
so here is the current method in `patient.rb` for it: https://github.com/SaraAlert/SaraAlert/blob/master/app/models/patient.rb#L546

it looks like most of the additional logic here is to ensure that the assessment found is not nil and that the new symptom onset date to be set is indeed before the current one

there's actually a problem with the current implementation of this in that if an assessment is updated from symptomatic to asymp or records are marked as reviewed, the symptom onset date will not be updated accordingly

this new implementation solves this problem since the callback is executed upon create, update, and delete

the `refresh_symptom_onset` method is deleted in this PR along with the calls to it and is instead automatically updated when these callback methods are executed

Reply:
Makes sense!